app/Http/Controllers/PaymentController.php

```diff
@@ -2,9 +2,152 @@
 
 namespace App\Http\Controllers;
 
+use App\Models\Payment;
+use App\Models\Student;
 use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
 
 class PaymentController extends Controller
 {
-    //
+    /**
+     * create transaction.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function createTransaction()
+    {
+        $paystack = Payment::PAYMENT_PAYSTACK;
+        $flutter = Payment::PAYMENT_FLUTTER;
+
+        return view('payment', compact('flutter', 'paystack'));
+    }
+
+    /**
+     * Store Payment Details in DB
+     * 
+     * @return \Illuminate\Http\Response
+     */
+
+     public function startTransaction(Request $request)
+     {
+        // $this->authorize('create', Payment::class);
+
+        $request->validate([
+            'purpose' => 'required|max:200|String',
+            'amount' => 'required',
+            'on_behalf' => 'String|nullable'
+        ]);
+
+        $payment = new Payment();
+        $payment->user_id = Auth::id();
+        $payment->on_behalf = $request->input('on_behalf');
+        $payment->purpose = $request->input('purpose');
+        $payment->amount = $request->input('amount');
+        $payment->reference = Payment::ref();
+        $payment->payment_date = date('Y-m-d H:i:s');
+        $payment->payment_service =  $request->input('payment_service');
+        $payment->payment_status = Payment::STATUS_PENDING;
+        //dd($payment);
+        if ($payment->save()){
+            return redirect(route('initiate-payment', $payment, $id));// ->with('action-success','Exam has been successfully created');
```

Comment:
This view should do what is in this link  https://paystack.com/docs/payments/accept-payments/#popup

Just get the user email from the user model from their auth id
then Show what they are paying for and amount.
then Pay with Paystack